app/controllers/api/stateless/braintree/payment_methods_controller.rb

```diff
@@ -0,0 +1,45 @@
+# frozen_string_literal: true
+
+module PaymentBraintreeService
+  extend self
+
+  def payment_methods_for_member(member)
+    customer(member).payment_methods.order('created_at desc')
+  end
+
+  def payment_method_for_member(member:, id:)
+    customer(member).payment_methods.find(id)
+  end
+
+  def customer(member)
+    Payment::Braintree::Customer.find_by!(member_id: member.id)
+  end
+end
+
+module Api
+  module Stateless
+    module Braintree
+      class PaymentMethodsController < StatelessController
+        before_filter :authenticate_request!
+
+        def index
+          @payment_methods = PaymentBraintreeService.payment_methods_for_member(@current_member)
+        end
+
+        def destroy
+          @payment_method = PaymentBraintreeService.payment_method_for_member(member: @current_member, id: params[:id])
+
+          result = begin
+                     ::Braintree::PaymentMethod.delete(@payment_method.token)
+                   rescue ::Braintree::NotFoundError
+                     @payment_method.destroy
+                   end
+
+
+          @payment_method.destroy if result.success?
```

Comment:
If you end up rescuing from `::Braintree::NotFoundError`, `result` will be `@payment_method`, because that's what will get returned from `@payment_method.destroy`. It doesn't respond to `success?`, so if that happens, this line will throw ```undefined method `success?' for #<Payment::Braintree::PaymentMethod:0x005560d6a71570>```